module_utils/kafka_manager.py

```diff
@@ -597,14 +597,15 @@ def update_topics_partitions(self, topics):
             total_replica = len(replicas)
             old_partition = self.get_total_partitions_for_topic(topic_name)
             assignments = []
-            for _new_partition in range(partitions - old_partition):
-                assignment = []
-                for _replica in range(total_replica):
-                    assignment.append(next(brokers_iterator))
-                assignments.append(assignment)
-            topics_assignments.append(
-                (topic_name, (partitions, assignments))
-            )
+            if partitions > old_partition:
```

Comment:
We already do this check before in `is_topics_partitions_need_update` https://github.com/StephenSorriaux/ansible-kafka-admin/blob/f2c6461145386b7d636c80ddcf2d84eca4daa9e9/module_utils/kafka_manager.py#L548
It's seem that the bug your encounter come from here https://github.com/StephenSorriaux/ansible-kafka-admin/blob/f2c6461145386b7d636c80ddcf2d84eca4daa9e9/module_utils/kafka_manager.py#L1304 where i think need to be filtered like here https://github.com/StephenSorriaux/ansible-kafka-admin/blob/f2c6461145386b7d636c80ddcf2d84eca4daa9e9/module_utils/kafka_manager.py#L1288
```python
                self.update_topics_partitions({
                    topic['name']: topic['partitions']
                    for topic in topics if (topic['name']
                                            in topics_partition_need_update)
                })
```

Reply:
Good point, changed to this one